src/vs/vscode.d.ts

```diff
@@ -2972,6 +2972,37 @@ declare namespace vscode {
 	}
 
 	/**
+	 * An individual terminal instance within the integrated terminal.
+	 */
+	export interface Terminal {
+
```

Comment:
add `name` as readonly so people can access that data they have passed in the create call